OSDOCS-12424#Automate LSO cleanup process (GA) (#86288)
openshift/openshift-docs: modules/persistent-storage-local-removing-devices.adoc

.Procedure

> To delete LVs or LVSs:

Reviewer: ```suggestion
To delete LVs or LVSs, complete the following steps:
```
Lead-in sentences should be complete sentences, per the SSG: https://redhat-documentation.github.io/supplementary-style-guide/#lead-in-sentences

> . If there are any bound PVs, you need to delete the corresponding persistent volume claims (PVCs) to release the PVs:

Reviewer: ```suggestion
. If there are any bound PVs, take the following steps to delete the corresponding persistent volume claims (PVCs) to release the PVs:
```

.. To find bound PVs, run the following command:
+
[source, terminal]
----
$ oc get pv

NAME   CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM                 STORAGECLASS   VOLUMEATTRIBUTESCLASS   REASON   AGE
pv1    5Gi        RWO            Retain           Available                         slow           <unset>                          28s
pv2    30Gi       RWX            Retain           Bound       openshift/storage     my-sc          <unset>                          168d
----
+

> Bound PVs have a status of "Bound" and their corresponding PVCs appear in the "CLAIM" column. In the preceding example, the second PV is bound, and its PVC is "openshift/storage".

Reviewer: ```suggestion
Bound PVs have a status of `Bound` and their corresponding PVCs appear in the `CLAIM` column. In the preceding example, the second PV is bound, and its PVC is `openshift/storage`.
```
According to the OCP docs guidelines, I think these terms should be in backticks instead of double-quotation marks: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference. I suggested a few similar changes in other places in this file.

.. Delete corresponding PVCs of bound PVs by running the following command:
+
[source,terminal]
----
$ oc delete pvc <name>
----
+ 

> In this example, you would delete PVC "openshift/storage".

Reviewer: ```suggestion
In this example, you would delete the `openshift/storage` PVC.
```

. If any PV owned by the LV or LVS has a "Retain" reclaim policy, back up any important data, and then delete the PV:

> .. To find PVs with "Retain" reclaim policy, run the following command:

Reviewer: ```suggestion
.. To find PVs with `Retain` reclaim policy, run the following command:
```

+
[source,terminal]
----
$ oc get pv

> 

Reviewer: Consider separating the command from the output code block.

NAME                CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM   STORAGECLASS   REASON   AGE
local-pv-1cec77cf   100Gi      RWO            Delete           Available           local-sc                88m
local-pv-2ef7cd2a   100Gi      RWO            Delete           Available           local-sc                82m
local-pv-3fa1c73    100Gi      RWO            Retain           Available           local-sc                48m]

> ----

Reviewer 2: I am not sure about `Retain` policy. 
@dobsonj Could you help to double confirm how we process this `Retain` policy?

+
In this example, the third PV has a retain reclaim policy.

.. Back up any important data on this volume.

.. Delete the PV by running the following command:
+
[source,terminal]
----
$ oc delete pv <name>
----
+

> In this example, the PV name is "local-pv-3fa1c73".

Reviewer: ```suggestion
In this example, the PV name is `local-pv-3fa1c73`.
```

.Procedure

To delete LVs or LVSs, complete the following steps:

. If there are any bound PVs, delete the corresponding persistent volume claims (PVCs) to release the PVs:

.. To find bound PVs, run the following command:
+
[source, terminal]
----
$ oc get pv
----
+
.Example output
[source,terminal]
----
NAME   CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM                 STORAGECLASS   VOLUMEATTRIBUTESCLASS   REASON   AGE
pv1    5Gi        RWO            Retain           Available                         slow           <unset>                          28s

> pv2    30Gi       RWX            Retain           Bound       openshift/storage     my-sc          <unset>                          168d

Reviewer 3: I would suggest using PV names like `local-pv-1cec77cf` in this example to stick with the naming scheme LSO uses when creating them.

----
+ 
In this example, you would delete PVC `openshift/storage`.

. Delete the LVs or LVSs by running the applicable following command:
+
.Command for deleting LV
+
[source,terminal]
----
$ oc delete lv <name>
----
+
or
+
.Command for deleting LVS
[source,terminal]
----
$ oc delete lvs <name>
----

. If any PV owned by the LV or LVS has a `Retain` reclaim policy, back up any important data, and then delete the PV:
+
[NOTE]
====
PVs with a `Delete` policy are automatically deleted when you delete the LVs or LVS.
====
+
.. To find PVs with `Retain` reclaim policy, run the following command:
+
[source,terminal]
----
$ oc get pv
----
+
.Example output
[source,terminal]
----
NAME                CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS      CLAIM                STORAGECLASS   REASON   AGE
local-pv-1cec77cf   30Gi       RWX            Retain           Available                        my-sc                   168d
----
+
In this example,  PV `local-pv-1cec77cf` has a `Retain` reclaim policy and needs to be manually deleted.

.. Back up any important data on this volume.

.. Delete the PV by running the following command:
+
[source,terminal]
----
$ oc delete pv <name>
----
+

> In this example, delete PV local-pv-3fa1c73`.

Reviewer 4: don't we want to delete local-pv-1cec77cf in that case?
@dobsonj

Author: Yes! Missed that. It's fixed now.
